tx/textual/valuerenderer/repeated.go

```diff
@@ -0,0 +1,83 @@
+package valuerenderer
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+type repeatedValueRenderer struct {
+	tr      *Textual
+	msgDesc protoreflect.MessageDescriptor
+	kind    string
+	vr      ValueRenderer
+}
+
+func NewRepeatedValueRenderer(t *Textual, msgDesc protoreflect.MessageDescriptor, kind string, v ValueRenderer) ValueRenderer {
+	return &repeatedValueRenderer{tr: t, msgDesc: msgDesc, kind: kind, vr: v}
+}
+
+func (mr *repeatedValueRenderer) header(len int) string {
+	name := mr.kind
+	if mr.msgDesc != nil {
+		name = string(mr.msgDesc.Name())
+	}
+	return fmt.Sprintf("%d %s", len, formatFieldName(name))
+}
+
+func (mr *repeatedValueRenderer) Format(ctx context.Context, v protoreflect.Value) ([]Screen, error) {
+	l := v.List()
+
+	if l == nil {
+		return nil, fmt.Errorf("non-List value")
+	}
+
+	if l.Len() == 0 {
+		return []Screen{}, nil
```

Comment:
Add a note that this should never occur, since a zero-length list is the default value for a repeated field, so it will be skipped by the enclosing message renderer.